src/commands/downloadDevContainer.ts

```diff
@@ -0,0 +1,63 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { WorkspaceFolder } from "vscode";
+import { IActionContext } from "vscode-azureextensionui";
+import { getDevContainerName } from '../downloadAzureProject/setupProjectFolder';
+import { ext } from '../extensionVariables';
+import { localize } from "../localize";
+import { ProductionSlotTreeItem } from '../tree/ProductionSlotTreeItem';
+import { SlotTreeItemBase } from '../tree/SlotTreeItemBase';
+import { requestUtils } from '../utils/requestUtils';
+import { selectWorkspaceFolder } from '../utils/workspace';
+import { tryGetFunctionProjectRoot } from './createNewProject/verifyIsProject';
+
+
+export async function downloadDevContainer(context: IActionContext, node?: SlotTreeItemBase): Promise<void> {
```

Comment:
because this is a helper method, we should make it where the `node` parameter is always required, rather than being optional. This would put the burden of verification and reassignment of `node` on the callers the helper method doesn't need an if/else associated with it.